<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Steven Yu" />

<meta name="date" content="2025-08-01" />

<title>Final Analysis</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">multiQC_cut&tag</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/infurnoheat/WardLab">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Final Analysis</h1>
<h4 class="author">Steven Yu</h4>
<h4 class="date">2025-08-01</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-08-04
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>WardLab/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250701code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250701)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250701code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250701)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcominfurnoheatWardLabtreefad9b2f85b6c48bbc1d4e0e7d279e2c503ab5201targetblankfad9b2fa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/infurnoheat/WardLab/tree/fad9b2f85b6c48bbc1d4e0e7d279e2c503ab5201" target="_blank">fad9b2f</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcominfurnoheatWardLabtreefad9b2f85b6c48bbc1d4e0e7d279e2c503ab5201targetblankfad9b2fa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/infurnoheat/WardLab/tree/fad9b2f85b6c48bbc1d4e0e7d279e2c503ab5201" target="_blank">fad9b2f</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  000012.png
    Untracked:  000016.png
    Untracked:  2ed61fa2-2e95-413f-a715-d4a6a74f5e82.png
    Untracked:  428606d7-cff3-4277-8cb5-3afcc33a2a2e.png
    Untracked:  MW_rotation_presentation.pptx
    Untracked:  data/RUVs_DEG.tsv
    Untracked:  data/bam_final/
    Untracked:  data/bam_no_multi/
    Untracked:  data/cutoff_narrow.tsv
    Untracked:  data/cutoff_narrow.tsv.txt
    Untracked:  data/cutoffs.tsv
    Untracked:  data/metadata.txt
    Untracked:  data/multiqc_data_trim/
    Untracked:  data/peaks/
    Untracked:  data/sample_info.tsv
    Untracked:  data/sams/

Unstaged changes:
    Modified:   README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/final_analysis.Rmd</code>) and
HTML (<code>docs/final_analysis.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/infurnoheat/WardLab/blob/fad9b2f85b6c48bbc1d4e0e7d279e2c503ab5201/analysis/final_analysis.Rmd" target="_blank">fad9b2f</a>
</td>
<td>
infurnoheat
</td>
<td>
2025-08-04
</td>
<td>
wflow_publish("analysis/final_analysis.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/infurnoheat/WardLab/f4826e12d1395e4a0e064036c446f0e0eb91e168/docs/final_analysis.html" target="_blank">f4826e1</a>
</td>
<td>
infurnoheat
</td>
<td>
2025-08-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/infurnoheat/WardLab/blob/f9fe741cd0a3335155dc369fade59fdffb267155/analysis/final_analysis.Rmd" target="_blank">f9fe741</a>
</td>
<td>
infurnoheat
</td>
<td>
2025-08-01
</td>
<td>
wflow_publish("analysis/final_analysis.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="final-anaylsis" class="section level2">
<h2>Final Anaylsis</h2>
<div id="loading-packages" class="section level4">
<h4>Loading Packages</h4>
<pre class="r"><code>library(tidyverse)
library(readr)
library(edgeR)
library(ComplexHeatmap)
library(data.table)
library(dplyr)
library(stringr)
library(ggplot2)
library(viridis)
library(DT)
library(kableExtra)
library(genomation)
library(GenomicRanges)
library(chromVAR) ## For FRiP analysis and differential analysis
library(DESeq2) ## For differential analysis section
library(ggpubr) ## For customizing figures
library(corrplot) ## For correlation plot
library(ggpmisc)
library(gcplyr)
library(Rsubread)
library(limma)
library(ggrastr)
library(cowplot)
library(smplot2)
library(ggVennDiagram)</code></pre>
</div>
<div id="data-initialization" class="section level4">
<h4>Data Initialization</h4>
<pre class="r"><code>sampleinfo &lt;- read_delim(&quot;data/sample_info.tsv&quot;, delim = &quot;\t&quot;)</code></pre>
</div>
<div id="functions" class="section level4">
<h4>Functions</h4>
<pre class="r"><code>drug_pal &lt;- c(&quot;#8B006D&quot;,&quot;#DF707E&quot;,&quot;#F1B72B&quot;, &quot;#3386DD&quot;,&quot;#707031&quot;,&quot;#41B333&quot;)
pca_plot &lt;-
  function(df,
           col_var = NULL,
           shape_var = NULL,
           text_var = NULL,
           title = &quot;&quot;) {
    ggplot(df, aes_string(x = &quot;PC1&quot;, y = &quot;PC2&quot;)) +
      geom_point(aes_string(color = col_var, shape = shape_var), size = 5) +
      labs(title = title, x = &quot;PC 1&quot;, y = &quot;PC 2&quot;) +
      ggrepel::geom_text_repel(label = text_var, vjust = -.5, max.overlaps = 30) +
      scale_color_manual(values = c(
        &quot;#8B006D&quot;,
        &quot;#DF707E&quot;,
        &quot;#F1B72B&quot;,
        &quot;#3386DD&quot;,
        &quot;#707031&quot;,
        &quot;#41B333&quot;
      ))
  }
pca_var_plot &lt;- function(pca) {
  # x: class == prcomp
  pca.var &lt;- pca$sdev ^ 2
  pca.prop &lt;- pca.var / sum(pca.var)
  var.plot &lt;-
    qplot(PC, prop, data = data.frame(PC = 1:length(pca.prop),
                                      prop = pca.prop)) +
    labs(title = &#39;Variance contributed by each PC&#39;,
         x = &#39;PC&#39;, y = &#39;Proportion of variance&#39;)
  plot(var.plot)
}

calc_pca &lt;- function(x) {
  # Performs principal components analysis with prcomp
  # x: a sample-by-gene numeric matrix
  prcomp(x, scale. = TRUE, retx = TRUE)
}

get_regr_pval &lt;- function(mod) {
  # Returns the p-value for the Fstatistic of a linear model
  # mod: class lm
  stopifnot(class(mod) == &quot;lm&quot;)
  fstat &lt;- summary(mod)$fstatistic
  pval &lt;- 1 - pf(fstat[1], fstat[2], fstat[3])
  return(pval)
}

plot_versus_pc &lt;- function(df, pc_num, fac) {
  # df: data.frame
  # pc_num: numeric, specific PC for plotting
  # fac: column name of df for plotting against PC
  pc_char &lt;- paste0(&quot;PC&quot;, pc_num)
  # Calculate F-statistic p-value for linear model
  pval &lt;- get_regr_pval(lm(df[, pc_char] ~ df[, fac]))
  if (is.numeric(df[, f])) {
    ggplot(df, aes_string(x = f, y = pc_char)) + geom_point() +
      geom_smooth(method = &quot;lm&quot;) + labs(title = sprintf(&quot;p-val: %.2f&quot;, pval))
  } else {
    ggplot(df, aes_string(x = f, y = pc_char)) + geom_boxplot() +
      labs(title = sprintf(&quot;p-val: %.2f&quot;, pval))
  }
}
x_axis_labels = function(labels, every_nth = 1, ...) {
  axis(side = 1,
       at = seq_along(labels),
       labels = F)
  text(
    x = (seq_along(labels))[seq_len(every_nth) == 1],
    y = par(&quot;usr&quot;)[3] - 0.075 * (par(&quot;usr&quot;)[4] - par(&quot;usr&quot;)[3]),
    labels = labels[seq_len(every_nth) == 1],
    xpd = TRUE,
    ...
  )
}

volcanosig &lt;- function(df, psig.lvl) {
    df &lt;- df %&gt;% 
    mutate(threshold = ifelse(adj.P.Val &gt; psig.lvl, &quot;A&quot;, ifelse(adj.P.Val &lt;= psig.lvl &amp; logFC&lt;=0,&quot;B&quot;,&quot;C&quot;)))
      # ifelse(adj.P.Val &lt;= psig.lvl &amp; logFC &gt;= 0,&quot;B&quot;, &quot;C&quot;)))
    ##This is where I could add labels, but I have taken out
    # df &lt;- df %&gt;% mutate(genelabels = &quot;&quot;)
    # df$genelabels[1:topg] &lt;- df$rownames[1:topg]
    
  ggplot(df, aes(x=logFC, y=-log10(P.Value))) + 
    ggrastr::geom_point_rast(aes(color=threshold))+
    # geom_text_repel(aes(label = genelabels), segment.curvature = -1e-20,force = 1,size=2.5,
    # arrow = arrow(length = unit(0.015, &quot;npc&quot;)), max.overlaps = Inf) +
    #geom_hline(yintercept = -log10(psig.lvl))+
    xlab(expression(&quot;Log&quot;[2]*&quot; FC&quot;))+
    ylab(expression(&quot;-log&quot;[10]*&quot;P Value&quot;))+
    scale_color_manual(values = c(&quot;black&quot;, &quot;red&quot;,&quot;blue&quot;))+
    theme_cowplot()+
    ylim(0,25)+
    xlim(-6,6)+
    theme(legend.position = &quot;none&quot;,
              plot.title = element_text(size = rel(1.5), hjust = 0.5),
              axis.title = element_text(size = rel(0.8))) 
}</code></pre>
</div>
<div id="final-analysis" class="section level3">
<h3>Final Analysis</h3>
<pre class="r"><code>peak_ct &lt;- read_delim(&quot;data/peaks/peaks_cts_FINAL.txt&quot;, delim = &quot;\t&quot;)
H3K27ac_peaks &lt;- read_delim(&quot;data/peaks/H3K27ac_FINAl_results.tsv&quot;,delim = &quot;\t&quot;)
H3K27me3_peaks &lt;- read_delim(&quot;data/peaks/H3K27me3_FINAL_results.tsv&quot;,delim = &quot;\t&quot;)
H3K36me3_peaks &lt;- read_delim(&quot;data/peaks/H3K36me3_FINAL_results.tsv&quot;,delim = &quot;\t&quot;)
H3K9me3_peaks &lt;- read_delim(&quot;data/peaks/H3K9me3_FINAL_results.tsv&quot;,delim = &quot;\t&quot;)

all_peak_final &lt;- rbind(H3K27ac_peaks, H3K27me3_peaks, H3K36me3_peaks, H3K9me3_peaks)

all_peak_final &lt;- all_peak_final %&gt;%
  dplyr::select(Sample, Total_Reads, Fragments, Reads_in_Peaks, FRiP) %&gt;%
  left_join(.,sampleinfo, by=c(&quot;Sample&quot;=&quot;Library ID&quot;)) %&gt;%
  left_join(.,peak_ct, by=c(&quot;Sample&quot;=&quot;Sample&quot;))
all_peak_final &lt;- all_peak_final[(!all_peak_final$Treatment %in% &quot;5FU&quot;),]</code></pre>
<pre class="r"><code>all_peak_final %&gt;% 
   ggplot(.,aes(x=Sample, y=Count,fill=Histone_Mark))+
   geom_col()+
   ylab(&quot;Count&quot;)+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle(&quot;Peak number for all samples&quot;)+ 
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  scale_y_continuous( expand = expansion(mult = c(0, .1)))</code></pre>
<p><img src="figure/final_analysis.Rmd/peak%20ct%20final-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-peak-ct-final-1">
Past versions of "peak ct final-1.png"
</button>
</p>
<div id="fig-no-spaces-peak-ct-final-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/infurnoheat/WardLab/blob/f4826e12d1395e4a0e064036c446f0e0eb91e168/docs/figure/final_analysis.Rmd/peak ct final-1.png" target="_blank">f4826e1</a>
</td>
<td>
infurnoheat
</td>
<td>
2025-08-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>all_peak_final %&gt;% 
  ggplot(., aes (x=Treatment, y = Count, fill = Histone_Mark))+
  geom_boxplot()+
   ylab(&quot;Count&quot;)+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle(&quot;Peak count across histones&quot;)</code></pre>
<p><img src="figure/final_analysis.Rmd/peak%20ct%20grouptx%20final-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-peak-ct-grouptx-final-1">
Past versions of "peak ct grouptx final-1.png"
</button>
</p>
<div id="fig-no-spaces-peak-ct-grouptx-final-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/infurnoheat/WardLab/blob/f4826e12d1395e4a0e064036c446f0e0eb91e168/docs/figure/final_analysis.Rmd/peak ct grouptx final-1.png" target="_blank">f4826e1</a>
</td>
<td>
infurnoheat
</td>
<td>
2025-08-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>all_peak_final %&gt;% 
  ggplot(., aes (x=Timepoint, y = Count, fill = Histone_Mark))+
  geom_boxplot()+
   ylab(&quot;Count&quot;)+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle(&quot;Peak count across histones&quot;)</code></pre>
<p><img src="figure/final_analysis.Rmd/peak%20ct%20grouptime%201e3_narrow-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-peak-ct-grouptime-1e3_narrow-1">
Past versions of "peak ct grouptime 1e3_narrow-1.png"
</button>
</p>
<div id="fig-no-spaces-peak-ct-grouptime-1e3_narrow-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/infurnoheat/WardLab/blob/f4826e12d1395e4a0e064036c446f0e0eb91e168/docs/figure/final_analysis.Rmd/peak ct grouptime 1e3_narrow-1.png" target="_blank">f4826e1</a>
</td>
<td>
infurnoheat
</td>
<td>
2025-08-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="tagging-questionable-libraries-by-frip" class="section level4">
<h4>Tagging Questionable Libraries by FRiP</h4>
<pre class="r"><code>questionable_frip = all_peak_final[(all_peak_final$FRiP &lt; 0.02),]
questionable_frip</code></pre>
<pre><code># A tibble: 0 × 10
# ℹ 10 variables: Sample &lt;chr&gt;, Total_Reads &lt;dbl&gt;, Fragments &lt;dbl&gt;,
#   Reads_in_Peaks &lt;dbl&gt;, FRiP &lt;dbl&gt;, Histone_Mark &lt;chr&gt;, Individual &lt;chr&gt;,
#   Treatment &lt;chr&gt;, Timepoint &lt;chr&gt;, Count &lt;dbl&gt;</code></pre>
</div>
</div>
<div id="feature-counts" class="section level3">
<h3>Feature Counts</h3>
<pre class="r"><code>H3K27ac_merged &lt;- read_delim(&quot;data/peaks/H3K27ac_FINAL_counts.txt&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    trim_ws = TRUE, skip = 1)
H3K27me3_merged &lt;- read_delim(&quot;data/peaks/H3K27me3_FINAL_counts.txt&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    trim_ws = TRUE, skip = 1)
H3K36me3_merged &lt;- read_delim(&quot;data/peaks/H3K36me3_FINAL_counts.txt&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    trim_ws = TRUE, skip = 1)
H3K9me3_merged &lt;- read_delim(&quot;data/peaks/H3K9me3_FINAL_counts.txt&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    trim_ws = TRUE, skip = 1)
rename_list &lt;- sampleinfo %&gt;% 
  mutate(stem= &quot;_nobl.bam&quot;) %&gt;% 
  mutate(prefix=paste0(&quot;/scratch/10819/styu/MW_multiQC/peaks/&quot;,Histone_Mark,&quot;/&quot;,Treatment,&quot;/&quot;,Timepoint,&quot;/&quot;)) %&gt;%
  mutate(oldname=paste0(prefix,`Library ID`,&quot;/&quot;,`Library ID`,stem)) %&gt;% 
  mutate(newname=paste0(Individual,&quot;_&quot;,Treatment,&quot;_&quot;,Timepoint)) %&gt;% 
  dplyr::select(oldname,newname)
rename_vec &lt;- setNames(rename_list$newname, rename_list$oldname)
names(H3K27ac_merged)[names(H3K27ac_merged) %in% names(rename_vec)] &lt;- rename_vec[names(H3K27ac_merged)[names(H3K27ac_merged) %in% names(rename_vec)]]
names(H3K27me3_merged)[names(H3K27me3_merged) %in% names(rename_vec)] &lt;- rename_vec[names(H3K27me3_merged)[names(H3K27me3_merged) %in% names(rename_vec)]]
names(H3K36me3_merged)[names(H3K36me3_merged) %in% names(rename_vec)] &lt;- rename_vec[names(H3K36me3_merged)[names(H3K36me3_merged) %in% names(rename_vec)]]
names(H3K9me3_merged)[names(H3K9me3_merged) %in% names(rename_vec)] &lt;- rename_vec[names(H3K9me3_merged)[names(H3K9me3_merged) %in% names(rename_vec)]]</code></pre>
</div>
<div id="h3k27ac-count-analysis" class="section level3">
<h3>H3K27ac Count Analysis</h3>
<pre class="r"><code>H3K27ac_merged_raw &lt;- H3K27ac_merged %&gt;% 
  dplyr::select(Geneid,contains(&quot;Ind&quot;)) %&gt;% 
  column_to_rownames(&quot;Geneid&quot;) %&gt;% 
  as.matrix()

H3K27ac_merged_lcpm &lt;- H3K27ac_merged %&gt;% 
  dplyr::select(Geneid,contains(&quot;Ind&quot;)) %&gt;% 
  column_to_rownames(&quot;Geneid&quot;) %&gt;% 
  cpm(., log = TRUE)
H3K27ac_merged_cor &lt;- H3K27ac_merged_lcpm %&gt;% 
  cor()

annomat &lt;- data.frame(sample=colnames(H3K27ac_merged_cor)) %&gt;% 
  separate_wider_delim(sample,delim=&quot;_&quot;,names=c(&quot;Ind&quot;,&quot;Treatment&quot;,&quot;Timepoint&quot;),cols_remove = FALSE) %&gt;% 
  mutate(Treatment=factor(Treatment, levels = c(&quot;VEH&quot;,&quot;5FU&quot;,&quot;DOX&quot;)),
         Timepoint=factor(Timepoint, levels =c(&quot;24T&quot;,&quot;24R&quot;,&quot;144R&quot;))) %&gt;% 
  column_to_rownames(&quot;sample&quot;)
heatmap_first &lt;- ComplexHeatmap::HeatmapAnnotation(df = annomat)

Heatmap(H3K27ac_merged_cor, 
        top_annotation = heatmap_first,
        column_title=&quot;Unfiltered log2cpm H3K27ac with Standard Merging&quot;)</code></pre>
<p><img src="figure/final_analysis.Rmd/H3K27ac%20HeatMap%201-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-H3K27ac-HeatMap-1-1">
Past versions of "H3K27ac HeatMap 1-1.png"
</button>
</p>
<div id="fig-no-spaces-H3K27ac-HeatMap-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/infurnoheat/WardLab/blob/f4826e12d1395e4a0e064036c446f0e0eb91e168/docs/figure/final_analysis.Rmd/H3K27ac HeatMap 1-1.png" target="_blank">f4826e1</a>
</td>
<td>
infurnoheat
</td>
<td>
2025-08-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="h3k27me3-count-analysis" class="section level3">
<h3>H3K27me3 Count Analysis</h3>
<pre class="r"><code>H3K27me3_merged_raw &lt;- H3K27me3_merged %&gt;% 
  dplyr::select(Geneid,contains(&quot;Ind&quot;)) %&gt;% 
  column_to_rownames(&quot;Geneid&quot;) %&gt;% 
  as.matrix()

H3K27me3_merged_lcpm &lt;- H3K27me3_merged %&gt;% 
  dplyr::select(Geneid,contains(&quot;Ind&quot;)) %&gt;% 
  column_to_rownames(&quot;Geneid&quot;) %&gt;% 
  cpm(., log = TRUE)
H3K27me3_merged_cor &lt;- H3K27me3_merged_lcpm %&gt;% 
  cor()

annomat &lt;- data.frame(sample=colnames(H3K27me3_merged_cor)) %&gt;% 
  separate_wider_delim(sample,delim=&quot;_&quot;,names=c(&quot;Ind&quot;,&quot;Treatment&quot;,&quot;Timepoint&quot;),cols_remove = FALSE) %&gt;% 
  mutate(Treatment=factor(Treatment, levels = c(&quot;VEH&quot;,&quot;5FU&quot;,&quot;DOX&quot;)),
         Timepoint=factor(Timepoint, levels =c(&quot;24T&quot;,&quot;24R&quot;,&quot;144R&quot;))) %&gt;% 
  column_to_rownames(&quot;sample&quot;)
heatmap_first &lt;- ComplexHeatmap::HeatmapAnnotation(df = annomat)

Heatmap(H3K27me3_merged_cor, 
        top_annotation = heatmap_first,
        column_title=&quot;Unfiltered log2cpm H3K27me3 with Standard Merging&quot;)</code></pre>
<p><img src="figure/final_analysis.Rmd/H3K27me3%20HeatMap%201-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-H3K27me3-HeatMap-1-1">
Past versions of "H3K27me3 HeatMap 1-1.png"
</button>
</p>
<div id="fig-no-spaces-H3K27me3-HeatMap-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/infurnoheat/WardLab/blob/f4826e12d1395e4a0e064036c446f0e0eb91e168/docs/figure/final_analysis.Rmd/H3K27me3 HeatMap 1-1.png" target="_blank">f4826e1</a>
</td>
<td>
infurnoheat
</td>
<td>
2025-08-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="h3k36me3-count-analysis" class="section level3">
<h3>H3K36me3 Count Analysis</h3>
<pre class="r"><code>H3K36me3_merged_raw &lt;- H3K36me3_merged %&gt;% 
  dplyr::select(Geneid,contains(&quot;Ind&quot;)) %&gt;% 
  column_to_rownames(&quot;Geneid&quot;) %&gt;% 
  as.matrix()

H3K36me3_merged_lcpm &lt;- H3K36me3_merged %&gt;% 
  dplyr::select(Geneid,contains(&quot;Ind&quot;)) %&gt;% 
  column_to_rownames(&quot;Geneid&quot;) %&gt;% 
  cpm(., log = TRUE)
H3K36me3_merged_cor &lt;- H3K36me3_merged_lcpm %&gt;% 
  cor()

annomat &lt;- data.frame(sample=colnames(H3K36me3_merged_cor)) %&gt;% 
  separate_wider_delim(sample,delim=&quot;_&quot;,names=c(&quot;Ind&quot;,&quot;Treatment&quot;,&quot;Timepoint&quot;),cols_remove = FALSE) %&gt;% 
  mutate(Treatment=factor(Treatment, levels = c(&quot;VEH&quot;,&quot;5FU&quot;,&quot;DOX&quot;)),
         Timepoint=factor(Timepoint, levels =c(&quot;24T&quot;,&quot;24R&quot;,&quot;144R&quot;))) %&gt;% 
  column_to_rownames(&quot;sample&quot;)
heatmap_first &lt;- ComplexHeatmap::HeatmapAnnotation(df = annomat)

Heatmap(H3K36me3_merged_cor, 
        top_annotation = heatmap_first,
        column_title=&quot;Unfiltered log2cpm H3K36me3 with Standard Merging&quot;)</code></pre>
<p><img src="figure/final_analysis.Rmd/H3K36me3%20HeatMap%201-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-H3K36me3-HeatMap-1-1">
Past versions of "H3K36me3 HeatMap 1-1.png"
</button>
</p>
<div id="fig-no-spaces-H3K36me3-HeatMap-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/infurnoheat/WardLab/blob/f4826e12d1395e4a0e064036c446f0e0eb91e168/docs/figure/final_analysis.Rmd/H3K36me3 HeatMap 1-1.png" target="_blank">f4826e1</a>
</td>
<td>
infurnoheat
</td>
<td>
2025-08-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="h3k9me3-count-analysis" class="section level3">
<h3>H3K9me3 Count Analysis</h3>
<pre class="r"><code>H3K9me3_merged_raw &lt;- H3K9me3_merged %&gt;% 
  dplyr::select(Geneid,contains(&quot;Ind&quot;)) %&gt;% 
  column_to_rownames(&quot;Geneid&quot;) %&gt;% 
  as.matrix()

H3K9me3_merged_lcpm &lt;- H3K9me3_merged %&gt;% 
  dplyr::select(Geneid,contains(&quot;Ind&quot;)) %&gt;% 
  column_to_rownames(&quot;Geneid&quot;) %&gt;% 
  cpm(., log = TRUE)
H3K9me3_merged_cor &lt;- H3K9me3_merged_lcpm %&gt;% 
  cor()

annomat &lt;- data.frame(sample=colnames(H3K9me3_merged_cor)) %&gt;% 
  separate_wider_delim(sample,delim=&quot;_&quot;,names=c(&quot;Ind&quot;,&quot;Treatment&quot;,&quot;Timepoint&quot;),cols_remove = FALSE) %&gt;% 
  mutate(Treatment=factor(Treatment, levels = c(&quot;VEH&quot;,&quot;5FU&quot;,&quot;DOX&quot;)),
         Timepoint=factor(Timepoint, levels =c(&quot;24T&quot;,&quot;24R&quot;,&quot;144R&quot;))) %&gt;% 
  column_to_rownames(&quot;sample&quot;)
heatmap_first &lt;- ComplexHeatmap::HeatmapAnnotation(df = annomat)

Heatmap(H3K9me3_merged_cor, 
        top_annotation = heatmap_first,
        column_title=&quot;Unfiltered log2cpm H3K9me3 with Standard Merging&quot;)</code></pre>
<p><img src="figure/final_analysis.Rmd/H3K9me3%20HeatMap%201-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-H3K9me3-HeatMap-1-1">
Past versions of "H3K9me3 HeatMap 1-1.png"
</button>
</p>
<div id="fig-no-spaces-H3K9me3-HeatMap-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/infurnoheat/WardLab/blob/f4826e12d1395e4a0e064036c446f0e0eb91e168/docs/figure/final_analysis.Rmd/H3K9me3 HeatMap 1-1.png" target="_blank">f4826e1</a>
</td>
<td>
infurnoheat
</td>
<td>
2025-08-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="fragment-analysis" class="section level3">
<h3>Fragment Analysis</h3>
<pre class="r"><code>all_peak_final %&gt;%
  mutate(Fragments=Fragments/1000000) %&gt;% 
  ggplot(., aes(x=interaction(Individual,Treatment,Timepoint), y=Fragments, fill=Treatment, group = Treatment))+
  geom_col()+
  geom_text(aes(y = 0,label = Sample), vjust = 0.2, size = 3, angle = 90)+
  theme_classic()+
  facet_wrap(~Histone_Mark)+
  ggtitle(&quot;Fragment count by histone and sample&quot;)+
  ylab(&quot;Count of Fragments * 10^6&quot;)+
  xlab(&quot;Samples&quot;)+
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  scale_y_continuous( expand = expansion(mult = c(0, .1)))</code></pre>
<p><img src="figure/final_analysis.Rmd/frag%20ct-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-frag-ct-1">
Past versions of "frag ct-1.png"
</button>
</p>
<div id="fig-no-spaces-frag-ct-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/infurnoheat/WardLab/blob/f4826e12d1395e4a0e064036c446f0e0eb91e168/docs/figure/final_analysis.Rmd/frag ct-1.png" target="_blank">f4826e1</a>
</td>
<td>
infurnoheat
</td>
<td>
2025-08-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>all_peak_final %&gt;%
  mutate(FRiP=FRiP * 100) %&gt;% 
  ggplot(., aes(x=interaction(Individual,Treatment,Timepoint), y=FRiP, fill=Treatment, group = Treatment))+
  geom_col()+
  geom_text(aes(y = 0,label = Sample), vjust = 0.2, size = 3, angle = 90)+
  theme_classic()+
  facet_wrap(~Histone_Mark)+
  ggtitle(&quot;Frip Percent by histone and sample&quot;)+
  ylab(&quot;Frip %&quot;)+
  xlab(&quot;Samples&quot;)+
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  scale_y_continuous( expand = expansion(mult = c(0, .1)))</code></pre>
<p><img src="figure/final_analysis.Rmd/frag%20frip-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-frag-frip-1">
Past versions of "frag frip-1.png"
</button>
</p>
<div id="fig-no-spaces-frag-frip-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/infurnoheat/WardLab/blob/f4826e12d1395e4a0e064036c446f0e0eb91e168/docs/figure/final_analysis.Rmd/frag frip-1.png" target="_blank">f4826e1</a>
</td>
<td>
infurnoheat
</td>
<td>
2025-08-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="differential-analysis" class="section level3">
<h3>Differential Analysis</h3>
<div id="filtering-sex-chromosome" class="section level4">
<h4>Filtering Sex Chromosome</h4>
<pre class="r"><code>H3K27ac_merged_raw &lt;- H3K27ac_merged_raw[rowMeans(H3K27ac_merged_cor)&gt;0,]
H3K27ac_merged_raw &lt;- H3K27ac_merged_raw[!grepl(&quot;chrY&quot;,rownames(H3K27ac_merged_raw)),]

H3K27me3_merged_raw &lt;- H3K27me3_merged_raw[rowMeans(H3K27me3_merged_cor)&gt;0,]
H3K27me3_merged_raw &lt;- H3K27me3_merged_raw[!grepl(&quot;chrY&quot;,rownames(H3K27me3_merged_raw)),]

H3K36me3_merged_raw &lt;- H3K36me3_merged_raw[rowMeans(H3K36me3_merged_cor)&gt;0,]
H3K36me3_merged_raw &lt;- H3K36me3_merged_raw[!grepl(&quot;chrY&quot;,rownames(H3K36me3_merged_raw)),]

H3K9me3_merged_raw &lt;- H3K9me3_merged_raw[rowMeans(H3K9me3_merged_cor)&gt;0,]
H3K9me3_merged_raw &lt;- H3K9me3_merged_raw[!grepl(&quot;chrY&quot;,rownames(H3K9me3_merged_raw)),]</code></pre>
</div>
<div id="setting-up-matrix" class="section level4">
<h4>Setting up Matrix</h4>
<pre class="r"><code>H3K27ac_annomat &lt;- data.frame(timeset=colnames(H3K27ac_merged_raw)) %&gt;% 
  mutate(sample=timeset) %&gt;% 
  separate(timeset, into = c(&quot;ind&quot;,&quot;tx&quot;,&quot;time&quot;)) %&gt;% 
  mutate(tx=factor(tx, levels = c(&quot;VEH&quot;, &quot;DOX&quot;)),
         time=factor(time, levels =c(&quot;24T&quot;,&quot;24R&quot;,&quot;144R&quot;))) %&gt;%
  mutate(ind = gsub(&quot;Ind&quot;, &quot;&quot;, ind)) %&gt;%
  mutate(txtime = paste0(tx, &quot;_&quot;, time)) %&gt;%
  mutate(group = txtime)
H3K27ac_annomat$group &lt;- H3K27ac_annomat$group %&gt;%
  gsub(&quot;DOX_24T&quot;, &quot;1&quot;, .) %&gt;%
  gsub(&quot;DOX_24R&quot;, &quot;2&quot;, .) %&gt;%
  gsub(&quot;DOX_144R&quot;, &quot;3&quot;, .) %&gt;%
  gsub(&quot;VEH_24T&quot;, &quot;4&quot;, .) %&gt;%
  gsub(&quot;VEH_24R&quot;, &quot;5&quot;, .) %&gt;%
  gsub(&quot;VEH_144R&quot;, &quot;6&quot;, .)

H3K27me3_annomat &lt;- data.frame(timeset=colnames(H3K27me3_merged_raw)) %&gt;% 
  mutate(sample=timeset) %&gt;% 
  separate(timeset, into = c(&quot;ind&quot;,&quot;tx&quot;,&quot;time&quot;)) %&gt;% 
  mutate(tx=factor(tx, levels = c(&quot;VEH&quot;, &quot;DOX&quot;)),
         time=factor(time, levels =c(&quot;24T&quot;,&quot;24R&quot;,&quot;144R&quot;))) %&gt;%
  mutate(ind = gsub(&quot;Ind&quot;, &quot;&quot;, ind)) %&gt;%
  mutate(txtime = paste0(tx, &quot;_&quot;, time)) %&gt;%
  mutate(group = txtime)
H3K27me3_annomat$group &lt;- H3K27me3_annomat$group %&gt;%
  gsub(&quot;DOX_24T&quot;, &quot;1&quot;, .) %&gt;%
  gsub(&quot;DOX_24R&quot;, &quot;2&quot;, .) %&gt;%
  gsub(&quot;DOX_144R&quot;, &quot;3&quot;, .) %&gt;%
  gsub(&quot;VEH_24T&quot;, &quot;4&quot;, .) %&gt;%
  gsub(&quot;VEH_24R&quot;, &quot;5&quot;, .) %&gt;%
  gsub(&quot;VEH_144R&quot;, &quot;6&quot;, .)

H3K36me3_annomat &lt;- data.frame(timeset=colnames(H3K36me3_merged_raw)) %&gt;% 
  mutate(sample=timeset) %&gt;% 
  separate(timeset, into = c(&quot;ind&quot;,&quot;tx&quot;,&quot;time&quot;)) %&gt;% 
  mutate(tx=factor(tx, levels = c(&quot;VEH&quot;, &quot;DOX&quot;)),
         time=factor(time, levels =c(&quot;24T&quot;,&quot;24R&quot;,&quot;144R&quot;))) %&gt;%
  mutate(ind = gsub(&quot;Ind&quot;, &quot;&quot;, ind)) %&gt;%
  mutate(txtime = paste0(tx, &quot;_&quot;, time)) %&gt;%
  mutate(group = txtime)
H3K36me3_annomat$group &lt;- H3K36me3_annomat$group %&gt;%
  gsub(&quot;DOX_24T&quot;, &quot;1&quot;, .) %&gt;%
  gsub(&quot;DOX_24R&quot;, &quot;2&quot;, .) %&gt;%
  gsub(&quot;DOX_144R&quot;, &quot;3&quot;, .) %&gt;%
  gsub(&quot;VEH_24T&quot;, &quot;4&quot;, .) %&gt;%
  gsub(&quot;VEH_24R&quot;, &quot;5&quot;, .) %&gt;%
  gsub(&quot;VEH_144R&quot;, &quot;6&quot;, .)

H3K9me3_annomat &lt;- data.frame(timeset=colnames(H3K9me3_merged_raw)) %&gt;% 
  mutate(sample=timeset) %&gt;% 
  separate(timeset, into = c(&quot;ind&quot;,&quot;tx&quot;,&quot;time&quot;)) %&gt;% 
  mutate(tx=factor(tx, levels = c(&quot;VEH&quot;, &quot;DOX&quot;)),
         time=factor(time, levels =c(&quot;24T&quot;,&quot;24R&quot;,&quot;144R&quot;))) %&gt;%
  mutate(ind = gsub(&quot;Ind&quot;, &quot;&quot;, ind)) %&gt;%
  mutate(txtime = paste0(tx, &quot;_&quot;, time)) %&gt;%
  mutate(group = txtime)
H3K9me3_annomat$group &lt;- H3K9me3_annomat$group %&gt;%
  gsub(&quot;DOX_24T&quot;, &quot;1&quot;, .) %&gt;%
  gsub(&quot;DOX_24R&quot;, &quot;2&quot;, .) %&gt;%
  gsub(&quot;DOX_144R&quot;, &quot;3&quot;, .) %&gt;%
  gsub(&quot;VEH_24T&quot;, &quot;4&quot;, .) %&gt;%
  gsub(&quot;VEH_24R&quot;, &quot;5&quot;, .) %&gt;%
  gsub(&quot;VEH_144R&quot;, &quot;6&quot;, .)

dge_H3K27ac &lt;- edgeR::DGEList(counts = H3K27ac_merged_raw, group = H3K27ac_annomat$group, genes = row.names(H3K27ac_merged_raw))
dge_H3K27me3 &lt;- edgeR::DGEList(counts = H3K27me3_merged_raw, group = H3K27me3_annomat$group, genes = row.names(H3K27me3_merged_raw))
dge_H3K36me3 &lt;- edgeR::DGEList(counts = H3K36me3_merged_raw, group = H3K36me3_annomat$group, genes = row.names(H3K36me3_merged_raw))
dge_H3K9me3 &lt;- edgeR::DGEList(counts = H3K9me3_merged_raw, group = H3K9me3_annomat$group, genes = row.names(H3K9me3_merged_raw))

dge_H3K27ac &lt;- edgeR::calcNormFactors(dge_H3K27ac)
dge_H3K27me3 &lt;- edgeR::calcNormFactors(dge_H3K27me3)
dge_H3K36me3 &lt;- edgeR::calcNormFactors(dge_H3K36me3)
dge_H3K9me3 &lt;- edgeR::calcNormFactors(dge_H3K9me3)

mm_H3K27ac &lt;- model.matrix(~0 + H3K27ac_annomat$txtime)
colnames(mm_H3K27ac) &lt;- H3K27ac_annomat$txtime %&gt;% unique()

mm_H3K27me3 &lt;- model.matrix(~0 + H3K27me3_annomat$txtime)
colnames(mm_H3K27me3) &lt;- H3K27me3_annomat$txtime %&gt;% unique()

mm_H3K36me3 &lt;- model.matrix(~0 + H3K36me3_annomat$txtime)
colnames(mm_H3K36me3) &lt;- H3K36me3_annomat$txtime %&gt;% unique()

mm_H3K9me3 &lt;- model.matrix(~0 + H3K9me3_annomat$txtime)
colnames(mm_H3K9me3) &lt;- H3K9me3_annomat$txtime %&gt;% unique()</code></pre>
</div>
</div>
<div id="volcano-plots" class="section level3">
<h3>Volcano Plots</h3>
<div id="h3k27ac" class="section level4">
<h4>H3K27ac</h4>
<pre class="r"><code>y &lt;- voom(dge_H3K27ac, mm_H3K27ac, plot = FALSE)
corfit &lt;- duplicateCorrelation(y, mm_H3K27ac, block = H3K27ac_annomat$ind)
v &lt;- voom(dge_H3K27ac, mm_H3K27ac, block = H3K27ac_annomat$ind, correlation = corfit$consensus.correlation)
fit &lt;- lmFit(v, mm_H3K27ac, block = H3K27ac_annomat$ind, correlation = corfit$consensus.correlation)
cm &lt;- makeContrasts(
  DOX_24T.VEH_24T = DOX_24T-VEH_24T,
  DOX_24R.VEH_24R = DOX_24R-VEH_24R,
  DOX_144R.VEH_144R = DOX_144R-VEH_144R,
  levels = mm_H3K27ac)

fit2&lt;- contrasts.fit(fit, contrasts=cm)
efit2 &lt;- eBayes(fit2)

results = decideTests(efit2)

summary(results)</code></pre>
<pre><code>       DOX_24T.VEH_24T DOX_24R.VEH_24R DOX_144R.VEH_144R
Down             14891            7175                 9
NotSig          194933          206740            221757
Up               11944            7853                 2</code></pre>
<pre class="r"><code>plotSA(efit2, main=&quot;Mean-Variance trend for final model for H3K27ac&quot;)</code></pre>
<p><img src="figure/final_analysis.Rmd/H3K27ac%20volcano%202-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-H3K27ac-volcano-2-1">
Past versions of "H3K27ac volcano 2-1.png"
</button>
</p>
<div id="fig-no-spaces-H3K27ac-volcano-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/infurnoheat/WardLab/blob/f4826e12d1395e4a0e064036c446f0e0eb91e168/docs/figure/final_analysis.Rmd/H3K27ac volcano 2-1.png" target="_blank">f4826e1</a>
</td>
<td>
infurnoheat
</td>
<td>
2025-08-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>V.24T.top= topTable(efit2, coef=1, adjust.method=&quot;BH&quot;, number=Inf, sort.by=&quot;p&quot;)
V.24R.top= topTable(efit2, coef=2, adjust.method=&quot;BH&quot;, number=Inf, sort.by=&quot;p&quot;)
V.144R.top= topTable(efit2, coef=3, adjust.method=&quot;BH&quot;, number=Inf, sort.by=&quot;p&quot;)

H3K27ac_24T &lt;- volcanosig(V.24T.top, 0.05)+ ggtitle(&quot;DOX 24T&quot;)
H3K27ac_24R &lt;- volcanosig(V.24R.top, 0.05)+ ggtitle(&quot;DOX 24R&quot;)+ylab(&quot;&quot;)
H3K27ac_144R &lt;- volcanosig(V.144R.top, 0.05)+ ggtitle(&quot;DOX 144R&quot;)+ylab(&quot;&quot;)
plot_grid(H3K27ac_24T, H3K27ac_24R, H3K27ac_144R, rel_widths =c(1,1,1))</code></pre>
<p><img src="figure/final_analysis.Rmd/H3K27ac%20volcano%203-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-H3K27ac-volcano-3-1">
Past versions of "H3K27ac volcano 3-1.png"
</button>
</p>
<div id="fig-no-spaces-H3K27ac-volcano-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/infurnoheat/WardLab/blob/f4826e12d1395e4a0e064036c446f0e0eb91e168/docs/figure/final_analysis.Rmd/H3K27ac volcano 3-1.png" target="_blank">f4826e1</a>
</td>
<td>
infurnoheat
</td>
<td>
2025-08-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="h3k27me3" class="section level4">
<h4>H3K27me3</h4>
<pre class="r"><code>y &lt;- voom(dge_H3K27me3, mm_H3K27me3, plot = FALSE)
corfit &lt;- duplicateCorrelation(y, mm_H3K27me3, block = H3K27me3_annomat$ind)
v &lt;- voom(dge_H3K27me3, mm_H3K27me3, block = H3K27me3_annomat$ind, correlation = corfit$consensus.correlation)
fit &lt;- lmFit(v, mm_H3K27me3, block = H3K27me3_annomat$ind, correlation = corfit$consensus.correlation)
cm &lt;- makeContrasts(
  DOX_24T.VEH_24T = DOX_24T-VEH_24T,
  DOX_24R.VEH_24R = DOX_24R-VEH_24R,
  DOX_144R.VEH_144R = DOX_144R-VEH_144R,
  levels = mm_H3K27me3)

fit2&lt;- contrasts.fit(fit, contrasts=cm)
efit2 &lt;- eBayes(fit2)

results = decideTests(efit2)

summary(results)</code></pre>
<pre><code>       DOX_24T.VEH_24T DOX_24R.VEH_24R DOX_144R.VEH_144R
Down                20               6                 1
NotSig          163181          163273            163284
Up                  84               6                 0</code></pre>
<pre class="r"><code>plotSA(efit2, main=&quot;Mean-Variance trend for final model for H3K27me3&quot;)</code></pre>
<p><img src="figure/final_analysis.Rmd/H3K27me3%20volcano%202-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-H3K27me3-volcano-2-1">
Past versions of "H3K27me3 volcano 2-1.png"
</button>
</p>
<div id="fig-no-spaces-H3K27me3-volcano-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/infurnoheat/WardLab/blob/f4826e12d1395e4a0e064036c446f0e0eb91e168/docs/figure/final_analysis.Rmd/H3K27me3 volcano 2-1.png" target="_blank">f4826e1</a>
</td>
<td>
infurnoheat
</td>
<td>
2025-08-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>V.24T.top= topTable(efit2, coef=1, adjust.method=&quot;BH&quot;, number=Inf, sort.by=&quot;p&quot;)
V.24R.top= topTable(efit2, coef=2, adjust.method=&quot;BH&quot;, number=Inf, sort.by=&quot;p&quot;)
V.144R.top= topTable(efit2, coef=3, adjust.method=&quot;BH&quot;, number=Inf, sort.by=&quot;p&quot;)

H3K27me3_24T &lt;- volcanosig(V.24T.top, 0.05)+ ggtitle(&quot;DOX 24T&quot;)
H3K27me3_24R &lt;- volcanosig(V.24R.top, 0.05)+ ggtitle(&quot;DOX 24R&quot;)+ylab(&quot;&quot;)
H3K27me3_144R &lt;- volcanosig(V.144R.top, 0.05)+ ggtitle(&quot;DOX 144R&quot;)+ylab(&quot;&quot;)
plot_grid(H3K27me3_24T, H3K27me3_24R, H3K27me3_144R, rel_widths =c(1,1,1))</code></pre>
<p><img src="figure/final_analysis.Rmd/H3K27me3%20volcano%203-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-H3K27me3-volcano-3-1">
Past versions of "H3K27me3 volcano 3-1.png"
</button>
</p>
<div id="fig-no-spaces-H3K27me3-volcano-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/infurnoheat/WardLab/blob/f4826e12d1395e4a0e064036c446f0e0eb91e168/docs/figure/final_analysis.Rmd/H3K27me3 volcano 3-1.png" target="_blank">f4826e1</a>
</td>
<td>
infurnoheat
</td>
<td>
2025-08-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="h3k36me3" class="section level4">
<h4>H3K36me3</h4>
<pre class="r"><code>y &lt;- voom(dge_H3K36me3, mm_H3K36me3, plot = FALSE)
corfit &lt;- duplicateCorrelation(y, mm_H3K36me3, block = H3K36me3_annomat$ind)
v &lt;- voom(dge_H3K36me3, mm_H3K36me3, block = H3K36me3_annomat$ind, correlation = corfit$consensus.correlation)
fit &lt;- lmFit(v, mm_H3K36me3, block = H3K36me3_annomat$ind, correlation = corfit$consensus.correlation)
cm &lt;- makeContrasts(
  DOX_24T.VEH_24T = DOX_24T-VEH_24T,
  DOX_24R.VEH_24R = DOX_24R-VEH_24R,
  DOX_144R.VEH_144R = DOX_144R-VEH_144R,
  levels = mm_H3K36me3)

fit2&lt;- contrasts.fit(fit, contrasts=cm)
efit2 &lt;- eBayes(fit2)

results = decideTests(efit2)

summary(results)</code></pre>
<pre><code>       DOX_24T.VEH_24T DOX_24R.VEH_24R DOX_144R.VEH_144R
Down              1461             189                 0
NotSig          187538          189577            190076
Up                1077             310                 0</code></pre>
<pre class="r"><code>plotSA(efit2, main=&quot;Mean-Variance trend for final model for H3K36me3&quot;)</code></pre>
<p><img src="figure/final_analysis.Rmd/H3K36me3%20volcano%202-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-H3K36me3-volcano-2-1">
Past versions of "H3K36me3 volcano 2-1.png"
</button>
</p>
<div id="fig-no-spaces-H3K36me3-volcano-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/infurnoheat/WardLab/blob/f4826e12d1395e4a0e064036c446f0e0eb91e168/docs/figure/final_analysis.Rmd/H3K36me3 volcano 2-1.png" target="_blank">f4826e1</a>
</td>
<td>
infurnoheat
</td>
<td>
2025-08-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>V.24T.top= topTable(efit2, coef=1, adjust.method=&quot;BH&quot;, number=Inf, sort.by=&quot;p&quot;)
V.24R.top= topTable(efit2, coef=2, adjust.method=&quot;BH&quot;, number=Inf, sort.by=&quot;p&quot;)
V.144R.top= topTable(efit2, coef=3, adjust.method=&quot;BH&quot;, number=Inf, sort.by=&quot;p&quot;)

H3K36me3_24T &lt;- volcanosig(V.24T.top, 0.05)+ ggtitle(&quot;DOX 24T&quot;)
H3K36me3_24R &lt;- volcanosig(V.24R.top, 0.05)+ ggtitle(&quot;DOX 24R&quot;)+ylab(&quot;&quot;)
H3K36me3_144R &lt;- volcanosig(V.144R.top, 0.05)+ ggtitle(&quot;DOX 144R&quot;)+ylab(&quot;&quot;)
plot_grid(H3K36me3_24T, H3K36me3_24R, H3K36me3_144R, rel_widths =c(1,1,1))</code></pre>
<p><img src="figure/final_analysis.Rmd/H3K36me3%20volcano%203-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-H3K36me3-volcano-3-1">
Past versions of "H3K36me3 volcano 3-1.png"
</button>
</p>
<div id="fig-no-spaces-H3K36me3-volcano-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/infurnoheat/WardLab/blob/f4826e12d1395e4a0e064036c446f0e0eb91e168/docs/figure/final_analysis.Rmd/H3K36me3 volcano 3-1.png" target="_blank">f4826e1</a>
</td>
<td>
infurnoheat
</td>
<td>
2025-08-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="h3k9me3" class="section level4">
<h4>H3K9me3</h4>
<pre class="r"><code>y &lt;- voom(dge_H3K9me3, mm_H3K9me3, plot = FALSE)
corfit &lt;- duplicateCorrelation(y, mm_H3K9me3, block = H3K9me3_annomat$ind)
v &lt;- voom(dge_H3K9me3, mm_H3K9me3, block = H3K9me3_annomat$ind, correlation = corfit$consensus.correlation)
fit &lt;- lmFit(v, mm_H3K9me3, block = H3K9me3_annomat$ind, correlation = corfit$consensus.correlation)
cm &lt;- makeContrasts(
  DOX_24T.VEH_24T = DOX_24T-VEH_24T,
  DOX_24R.VEH_24R = DOX_24R-VEH_24R,
  DOX_144R.VEH_144R = DOX_144R-VEH_144R,
  levels = mm_H3K9me3)

fit2&lt;- contrasts.fit(fit, contrasts=cm)
efit2 &lt;- eBayes(fit2)

results = decideTests(efit2)

summary(results)</code></pre>
<pre><code>       DOX_24T.VEH_24T DOX_24R.VEH_24R DOX_144R.VEH_144R
Down               833               1                 0
NotSig          215326          225399            225547
Up                9388             147                 0</code></pre>
<pre class="r"><code>plotSA(efit2, main=&quot;Mean-Variance trend for final model for H3K9me3&quot;)</code></pre>
<p><img src="figure/final_analysis.Rmd/H3K9me3%20volcano%202-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-H3K9me3-volcano-2-1">
Past versions of "H3K9me3 volcano 2-1.png"
</button>
</p>
<div id="fig-no-spaces-H3K9me3-volcano-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/infurnoheat/WardLab/blob/f4826e12d1395e4a0e064036c446f0e0eb91e168/docs/figure/final_analysis.Rmd/H3K9me3 volcano 2-1.png" target="_blank">f4826e1</a>
</td>
<td>
infurnoheat
</td>
<td>
2025-08-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>V.24T.top= topTable(efit2, coef=1, adjust.method=&quot;BH&quot;, number=Inf, sort.by=&quot;p&quot;)
V.24R.top= topTable(efit2, coef=2, adjust.method=&quot;BH&quot;, number=Inf, sort.by=&quot;p&quot;)
V.144R.top= topTable(efit2, coef=3, adjust.method=&quot;BH&quot;, number=Inf, sort.by=&quot;p&quot;)

H3K9me3_24T &lt;- volcanosig(V.24T.top, 0.05)+ ggtitle(&quot;DOX 24T&quot;)
H3K9me3_24R &lt;- volcanosig(V.24R.top, 0.05)+ ggtitle(&quot;DOX 24R&quot;)+ylab(&quot;&quot;)
H3K9me3_144R &lt;- volcanosig(V.144R.top, 0.05)+ ggtitle(&quot;DOX 144R&quot;)+ylab(&quot;&quot;)
plot_grid(H3K9me3_24T, H3K9me3_24R, H3K9me3_144R, rel_widths =c(1,1,1))</code></pre>
<p><img src="figure/final_analysis.Rmd/H3K9me3%20volcano%203-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-H3K9me3-volcano-3-1">
Past versions of "H3K9me3 volcano 3-1.png"
</button>
</p>
<div id="fig-no-spaces-H3K9me3-volcano-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/infurnoheat/WardLab/blob/f4826e12d1395e4a0e064036c446f0e0eb91e168/docs/figure/final_analysis.Rmd/H3K9me3 volcano 3-1.png" target="_blank">f4826e1</a>
</td>
<td>
infurnoheat
</td>
<td>
2025-08-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="pca-plots" class="section level3">
<h3>PCA Plots</h3>
<div id="h3k27ac-1" class="section level4">
<h4>H3K27ac</h4>
<pre class="r"><code>pca_H3K27ac &lt;- calc_pca(t(H3K27ac_merged_lcpm))
pca_var_plot(pca_H3K27ac)</code></pre>
<p><img src="figure/final_analysis.Rmd/H3K27ac%20PCA_var-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pca_H3K27ac &lt;- pca_H3K27ac$x %&gt;% cbind(., H3K27ac_annomat)
pca_plot(pca_H3K27ac, col_var = &quot;time&quot;, shape_var = &quot;tx&quot;, text_var = pca_H3K27ac$ind, title = &quot;H3K27ac lcpm PCA&quot;)</code></pre>
<p><img src="figure/final_analysis.Rmd/H3K27ac%20PCA-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="h3k27me3-1" class="section level4">
<h4>H3K27me3</h4>
<pre class="r"><code>pca_H3K27me3 &lt;- calc_pca(t(H3K27me3_merged_lcpm))
pca_var_plot(pca_H3K27me3)</code></pre>
<p><img src="figure/final_analysis.Rmd/H3K27me3%20PCA_var-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pca_H3K27me3 &lt;- pca_H3K27me3$x %&gt;% cbind(., H3K27me3_annomat)
pca_plot(pca_H3K27me3, col_var = &quot;time&quot;, shape_var = &quot;tx&quot;, text_var = pca_H3K27me3$ind, title = &quot;H3K27me3 lcpm PCA&quot;)</code></pre>
<p><img src="figure/final_analysis.Rmd/H3K27me3%20PCA-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="h3k36me3-1" class="section level4">
<h4>H3K36me3</h4>
<pre class="r"><code>pca_H3K36me3 &lt;- calc_pca(t(H3K36me3_merged_lcpm))
pca_var_plot(pca_H3K36me3)</code></pre>
<p><img src="figure/final_analysis.Rmd/H3K36me3%20PCA_var-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pca_H3K36me3 &lt;- pca_H3K36me3$x %&gt;% cbind(., H3K36me3_annomat)
pca_plot(pca_H3K36me3, col_var = &quot;time&quot;, shape_var = &quot;tx&quot;, text_var = pca_H3K36me3$ind, title = &quot;H3K36me3 lcpm PCA&quot;)</code></pre>
<p><img src="figure/final_analysis.Rmd/H3K36me3%20PCA-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="h3k9me3-1" class="section level4">
<h4>H3K9me3</h4>
<pre class="r"><code>pca_H3K9me3 &lt;- calc_pca(t(H3K9me3_merged_lcpm))
pca_var_plot(pca_H3K9me3)</code></pre>
<p><img src="figure/final_analysis.Rmd/H3K9me3%20PCA_var-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pca_H3K9me3 &lt;- pca_H3K9me3$x %&gt;% cbind(., H3K9me3_annomat)
pca_plot(pca_H3K9me3, col_var = &quot;time&quot;, shape_var = &quot;tx&quot;, text_var = pca_H3K9me3$ind, title = &quot;H3K9me3 lcpm PCA&quot;)</code></pre>
<p><img src="figure/final_analysis.Rmd/H3K9me3%20PCA-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="venn-diagrams" class="section level3">
<h3>Venn Diagrams</h3>
<div id="venn-set-up" class="section level4">
<h4>Venn Set Up</h4>
<pre class="r"><code>genes_H3K27ac_24T &lt;- H3K27ac_24T$data$genes[(H3K27ac_24T$data$adj.P.Val &lt; 0.05)]
genes_H3K27ac_24R &lt;- H3K27ac_24R$data$genes[(H3K27ac_24R$data$adj.P.Val &lt; 0.05)]
genes_H3K27ac_144R &lt;- H3K27ac_144R$data$genes[(H3K27ac_144R$data$adj.P.Val &lt; 0.05)]

genes_H3K27me3_24T &lt;- H3K27me3_24T$data$genes[(H3K27me3_24T$data$adj.P.Val &lt; 0.05)]
genes_H3K27me3_24R &lt;- H3K27me3_24R$data$genes[(H3K27me3_24R$data$adj.P.Val &lt; 0.05)]
genes_H3K27me3_144R &lt;- H3K27me3_144R$data$genes[(H3K27me3_144R$data$adj.P.Val &lt; 0.05)]

genes_H3K36me3_24T &lt;- H3K36me3_24T$data$genes[(H3K36me3_24T$data$adj.P.Val &lt; 0.05)]
genes_H3K36me3_24R &lt;- H3K36me3_24R$data$genes[(H3K36me3_24R$data$adj.P.Val &lt; 0.05)]
genes_H3K36me3_144R &lt;- H3K36me3_144R$data$genes[(H3K36me3_144R$data$adj.P.Val &lt; 0.05)]

genes_H3K9me3_24T &lt;- H3K9me3_24T$data$genes[(H3K9me3_24T$data$adj.P.Val &lt; 0.05)]
genes_H3K9me3_24R &lt;- H3K9me3_24R$data$genes[(H3K9me3_24R$data$adj.P.Val &lt; 0.05)]
genes_H3K9me3_144R &lt;- H3K9me3_144R$data$genes[(H3K9me3_144R$data$adj.P.Val &lt; 0.05)]</code></pre>
</div>
<div id="h3k27ac-venn-diagrams" class="section level4">
<h4>H3K27ac Venn Diagrams</h4>
<pre class="r"><code>ggVennDiagram(list(&quot;24T regions&quot;=genes_H3K27ac_24T,&quot;24R regions&quot;=genes_H3K27ac_24R, &quot;144R regions&quot;=genes_H3K27ac_144R))</code></pre>
<p><img src="figure/final_analysis.Rmd/H3K27ac%20Venn-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="h3k27me3-venn-diagrams" class="section level4">
<h4>H3K27me3 Venn Diagrams</h4>
<pre class="r"><code>ggVennDiagram(list(&quot;24T regions&quot;=genes_H3K27me3_24T,&quot;24R regions&quot;=genes_H3K27me3_24R, &quot;144R regions&quot;=genes_H3K27me3_144R))</code></pre>
<p><img src="figure/final_analysis.Rmd/H3K27me3%20Venn-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="h3k36me3-venn-diagrams" class="section level4">
<h4>H3K36me3 Venn Diagrams</h4>
<pre class="r"><code>ggVennDiagram(list(&quot;24T regions&quot;=genes_H3K36me3_24T,&quot;24R regions&quot;=genes_H3K36me3_24R, &quot;144R regions&quot;=genes_H3K36me3_144R))</code></pre>
<p><img src="figure/final_analysis.Rmd/H3K36me3%20Venn-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="h3k9me3-venn-diagrams" class="section level4">
<h4>H3K9me3 Venn Diagrams</h4>
<pre class="r"><code>ggVennDiagram(list(&quot;24T regions&quot;=genes_H3K9me3_24T,&quot;24R regions&quot;=genes_H3K9me3_24R, &quot;144R regions&quot;=genes_H3K9me3_144R))</code></pre>
<p><img src="figure/final_analysis.Rmd/H3K9me3%20Venn-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.5.1 (2025-06-13 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 11 x64 (build 26100)

Matrix products: default
  LAPACK version 3.12.1

locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ggVennDiagram_1.5.4         smplot2_0.2.5              
 [3] cowplot_1.2.0               ggrastr_1.0.2              
 [5] Rsubread_2.22.1             gcplyr_1.12.0              
 [7] ggpmisc_0.6.2               ggpp_0.5.9                 
 [9] corrplot_0.95               ggpubr_0.6.1               
[11] DESeq2_1.48.1               SummarizedExperiment_1.38.1
[13] Biobase_2.68.0              MatrixGenerics_1.20.0      
[15] matrixStats_1.5.0           chromVAR_1.30.1            
[17] GenomicRanges_1.60.0        GenomeInfoDb_1.44.1        
[19] IRanges_2.42.0              S4Vectors_0.46.0           
[21] BiocGenerics_0.54.0         generics_0.1.4             
[23] genomation_1.40.1           kableExtra_1.4.0           
[25] DT_0.33                     viridis_0.6.5              
[27] viridisLite_0.4.2           data.table_1.17.8          
[29] ComplexHeatmap_2.24.1       edgeR_4.6.3                
[31] limma_3.64.1                lubridate_1.9.4            
[33] forcats_1.0.0               stringr_1.5.1              
[35] dplyr_1.1.4                 purrr_1.1.0                
[37] readr_2.1.5                 tidyr_1.3.1                
[39] tibble_3.3.0                ggplot2_3.5.2              
[41] tidyverse_2.0.0             workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] splines_4.5.1               later_1.4.2                
  [3] BiocIO_1.18.0               bitops_1.0-9               
  [5] rpart_4.1.24                XML_3.99-0.18              
  [7] DirichletMultinomial_1.50.0 lifecycle_1.0.4            
  [9] pwalign_1.4.0               rstatix_0.7.2              
 [11] doParallel_1.0.17           rprojroot_2.1.0            
 [13] vroom_1.6.5                 MASS_7.3-65                
 [15] processx_3.8.6              lattice_0.22-7             
 [17] backports_1.5.0             magrittr_2.0.3             
 [19] Hmisc_5.2-3                 plotly_4.11.0              
 [21] sass_0.4.10                 rmarkdown_2.29             
 [23] jquerylib_0.1.4             yaml_2.3.10                
 [25] plotrix_3.8-4               httpuv_1.6.16              
 [27] DBI_1.2.3                   RColorBrewer_1.1-3         
 [29] abind_1.4-8                 RCurl_1.98-1.17            
 [31] nnet_7.3-20                 git2r_0.36.2               
 [33] circlize_0.4.16             GenomeInfoDbData_1.2.14    
 [35] ggrepel_0.9.6               seqLogo_1.74.0             
 [37] MatrixModels_0.5-4          svglite_2.2.1              
 [39] codetools_0.2-20            DelayedArray_0.34.1        
 [41] xml2_1.3.8                  tidyselect_1.2.1           
 [43] shape_1.4.6.1               UCSC.utils_1.4.0           
 [45] farver_2.1.2                base64enc_0.1-3            
 [47] GenomicAlignments_1.44.0    jsonlite_2.0.0             
 [49] GetoptLong_1.0.5            Formula_1.2-5              
 [51] survival_3.8-3              iterators_1.0.14           
 [53] systemfonts_1.2.3           foreach_1.5.2              
 [55] tools_4.5.1                 TFMPvalue_0.0.9            
 [57] Rcpp_1.1.0                  glue_1.8.0                 
 [59] gridExtra_2.3               SparseArray_1.8.1          
 [61] xfun_0.52                   withr_3.0.2                
 [63] fastmap_1.2.0               SparseM_1.84-2             
 [65] callr_3.7.6                 caTools_1.18.3             
 [67] digest_0.6.37               timechange_0.3.0           
 [69] R6_2.6.1                    mime_0.13                  
 [71] seqPattern_1.40.0           textshaping_1.0.1          
 [73] colorspace_2.1-1            Cairo_1.6-2                
 [75] gtools_3.9.5                RSQLite_2.4.2              
 [77] rtracklayer_1.68.0          httr_1.4.7                 
 [79] htmlwidgets_1.6.4           S4Arrays_1.8.1             
 [81] TFBSTools_1.46.0            whisker_0.4.1              
 [83] pkgconfig_2.0.3             gtable_0.3.6               
 [85] blob_1.2.4                  impute_1.82.0              
 [87] XVector_0.48.0              htmltools_0.5.8.1          
 [89] carData_3.0-5               pwr_1.3-0                  
 [91] clue_0.3-66                 scales_1.4.0               
 [93] png_0.1-8                   knitr_1.50                 
 [95] rstudioapi_0.17.1           tzdb_0.5.0                 
 [97] reshape2_1.4.4              rjson_0.2.23               
 [99] checkmate_2.3.2             curl_6.4.0                 
[101] zoo_1.8-14                  cachem_1.1.0               
[103] GlobalOptions_0.1.2         KernSmooth_2.23-26         
[105] vipor_0.4.7                 parallel_4.5.1             
[107] miniUI_0.1.2                foreign_0.8-90             
[109] restfulr_0.0.16             pillar_1.11.0              
[111] vctrs_0.6.5                 promises_1.3.3             
[113] car_3.1-3                   xtable_1.8-4               
[115] cluster_2.1.8.1             htmlTable_2.4.3            
[117] beeswarm_0.4.0              evaluate_1.0.4             
[119] cli_3.6.5                   locfit_1.5-9.12            
[121] compiler_4.5.1              Rsamtools_2.24.0           
[123] rlang_1.1.6                 crayon_1.5.3               
[125] ggsignif_0.6.4              labeling_0.4.3             
[127] ps_1.9.1                    ggbeeswarm_0.7.2           
[129] getPass_0.2-4               plyr_1.8.9                 
[131] fs_1.6.6                    stringi_1.8.7              
[133] gridBase_0.4-7              BiocParallel_1.42.1        
[135] Biostrings_2.76.0           lazyeval_0.2.2             
[137] quantreg_6.1                Matrix_1.7-3               
[139] BSgenome_1.76.0             patchwork_1.3.1            
[141] hms_1.1.3                   bit64_4.6.0-1              
[143] statmod_1.5.0               shiny_1.11.1               
[145] broom_1.0.9                 memoise_2.0.1              
[147] bslib_0.9.0                 bit_4.6.0                  
[149] polynom_1.4-1              </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
