---
title: "Peak Analysis"
author: "Steven Yu"
date: "2025-07-14"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo = TRUE,
	message = FALSE,
	warning = FALSE,
	dev = c("png","pdf")
)
```

## Peak Anaylsis

#### Loading Packages 
```{r load packages}
library(tidyverse)
library(readr)
library(edgeR)
library(ComplexHeatmap)
library(data.table)
library(dplyr)
library(stringr)
library(ggplot2)
library(viridis)
library(DT)
library(kableExtra)
library(genomation)
library(GenomicRanges)
library(chromVAR) ## For FRiP analysis and differential analysis
library(DESeq2) ## For differential analysis section
library(ggpubr) ## For customizing figures
library(corrplot) ## For correlation plot
library(ggpmisc)
library(gcplyr)
library(Rsubread)
```

#### Data Initialization
```{r init}
sampleinfo <- read_delim("data/sample_info.tsv", delim = "\t")
multiqc_gene_stats_trim <- read_delim("data/multiqc_data_trim/multiqc_general_stats.txt",delim = "\t")
multiqc_fastqc_trim <- read_delim("data/multiqc_data_trim/multiqc_fastqc.txt",delim = "\t")
```

#### Functions
```{r functions}
drug_pal <- c("#8B006D","#DF707E","#F1B72B", "#3386DD","#707031","#41B333")
pca_plot <-
  function(df,
           col_var = NULL,
           shape_var = NULL,
           title = "") {
    ggplot(df) + geom_point(aes_string(
      x = "PC1",
      y = "PC2",
      color = col_var,
      shape = shape_var
    ),
    size = 5) +
      labs(title = title, x = "PC 1", y = "PC 2") +
      scale_color_manual(values = c(
        "#8B006D",
        "#DF707E",
        "#F1B72B",
        "#3386DD",
        "#707031",
        "#41B333"
      ))
  }
pca_var_plot <- function(pca) {
  # x: class == prcomp
  pca.var <- pca$sdev ^ 2
  pca.prop <- pca.var / sum(pca.var)
  var.plot <-
    qplot(PC, prop, data = data.frame(PC = 1:length(pca.prop),
                                      prop = pca.prop)) +
    labs(title = 'Variance contributed by each PC',
         x = 'PC', y = 'Proportion of variance')
  plot(var.plot)
}

calc_pca <- function(x) {
  # Performs principal components analysis with prcomp
  # x: a sample-by-gene numeric matrix
  prcomp(x, scale. = TRUE, retx = TRUE)
}

get_regr_pval <- function(mod) {
  # Returns the p-value for the Fstatistic of a linear model
  # mod: class lm
  stopifnot(class(mod) == "lm")
  fstat <- summary(mod)$fstatistic
  pval <- 1 - pf(fstat[1], fstat[2], fstat[3])
  return(pval)
}

plot_versus_pc <- function(df, pc_num, fac) {
  # df: data.frame
  # pc_num: numeric, specific PC for plotting
  # fac: column name of df for plotting against PC
  pc_char <- paste0("PC", pc_num)
  # Calculate F-statistic p-value for linear model
  pval <- get_regr_pval(lm(df[, pc_char] ~ df[, fac]))
  if (is.numeric(df[, f])) {
    ggplot(df, aes_string(x = f, y = pc_char)) + geom_point() +
      geom_smooth(method = "lm") + labs(title = sprintf("p-val: %.2f", pval))
  } else {
    ggplot(df, aes_string(x = f, y = pc_char)) + geom_boxplot() +
      labs(title = sprintf("p-val: %.2f", pval))
  }
}
x_axis_labels = function(labels, every_nth = 1, ...) {
  axis(side = 1,
       at = seq_along(labels),
       labels = F)
  text(
    x = (seq_along(labels))[seq_len(every_nth) == 1],
    y = par("usr")[3] - 0.075 * (par("usr")[4] - par("usr")[3]),
    labels = labels[seq_len(every_nth) == 1],
    xpd = TRUE,
    ...
  )
}
```

### Peak Calling for Q=0.01 and Broad=0.1 with No Lambda

#### Data Initialization
```{r peak data}
peak_ct <- read_delim("data/peaks/peaks_cts.txt", delim = "\t")
H3K27ac_peaks <- read_delim("data/peaks/H3K27ac_final_results.tsv",delim = "\t")
H3K27me3_peaks <- read_delim("data/peaks/H3K27me3_final_results.tsv",delim = "\t")
H3K36me3_peaks <- read_delim("data/peaks/H3K36me3_final_results.tsv",delim = "\t")
H3K9me3_peaks <- read_delim("data/peaks/H3K9me3_final_results.tsv",delim = "\t")

all_peak <- rbind(H3K27ac_peaks, H3K27me3_peaks, H3K36me3_peaks, H3K9me3_peaks)

all_peak <- all_peak %>%
  dplyr::select(Sample, Total_Reads, Fragments, Reads_in_Peaks, FRiP) %>%
  left_join(.,sampleinfo, by=c("Sample"="Library ID")) %>%
  left_join(.,peak_ct, by=c("Sample"="Sample"))
all_peak <- all_peak[(!all_peak$Treatment %in% "5FU"),]
```

#### Peak Visualization
```{r peak ct, fig.width=14}
all_peak %>% 
   ggplot(.,aes(x=Sample, y=Count,fill=Histone_Mark))+
   geom_col()+
   ylab("Count")+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle("Peak number for all samples")+ 
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  scale_y_continuous( expand = expansion(mult = c(0, .1)))
```

```{r peak ct box}
all_peak %>% 
  ggplot(., aes (x=Histone_Mark, y = Count, fill = Histone_Mark))+
  geom_boxplot()+
   ylab("Count")+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle("Peak count across histones")
```

```{r peak ct grouptx}
all_peak %>% 
  ggplot(., aes (x=Treatment, y = Count, fill = Histone_Mark))+
  geom_boxplot()+
   ylab("Count")+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle("Peak count across histones")
```

```{r peak ct grouptime}
all_peak %>% 
  ggplot(., aes (x=Timepoint, y = Count, fill = Histone_Mark))+
  geom_boxplot()+
   ylab("Count")+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle("Peak count across histones")
```

### Peak Calling for Q=0.01 and Broad=0.1 with Lambda

#### Data Initialization
```{r peak data lq1e2b1e1}
peak_ct <- read_delim("data/peaks/peaks_cts_lq1e2b1e1.txt", delim = "\t")
H3K27ac_peaks <- read_delim("data/peaks/H3K27ac_final_results.tsv",delim = "\t")
H3K27me3_peaks <- read_delim("data/peaks/H3K27me3_lq1e2b1e1_results.tsv",delim = "\t")
H3K36me3_peaks <- read_delim("data/peaks/H3K36me3_lq1e2b1e1_results.tsv",delim = "\t")
H3K9me3_peaks <- read_delim("data/peaks/H3K9me3_lq1e2b1e1_results.tsv",delim = "\t")

all_peak_var <- rbind(H3K27ac_peaks, H3K27me3_peaks, H3K36me3_peaks, H3K9me3_peaks)

all_peak_var <- all_peak_var %>%
  dplyr::select(Sample, Total_Reads, Fragments, Reads_in_Peaks, FRiP) %>%
  left_join(.,sampleinfo, by=c("Sample"="Library ID")) %>%
  left_join(.,peak_ct, by=c("Sample"="Sample"))
all_peak_var <- all_peak_var[(!all_peak_var$Treatment %in% "5FU"),]
all_peak_var <- all_peak_var[(!all_peak_var$Histone_Mark %in% "H3K27ac"),]
```

#### Peak Visualization
```{r peak ct lq1e2b1e1, fig.width=14}
all_peak_var %>% 
   ggplot(.,aes(x=Sample, y=Count,fill=Histone_Mark))+
   geom_col()+
   ylab("Count")+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle("Peak number for all samples")+ 
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  scale_y_continuous( expand = expansion(mult = c(0, .1)))
```

```{r peak ct box lq1e2b1e1}
all_peak_var %>% 
  ggplot(., aes (x=Histone_Mark, y = Count, fill = Histone_Mark))+
  geom_boxplot()+
   ylab("Count")+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle("Peak count across histones")
```

```{r peak ct grouptx lq1e2b1e1}
all_peak_var %>% 
  ggplot(., aes (x=Treatment, y = Count, fill = Histone_Mark))+
  geom_boxplot()+
   ylab("Count")+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle("Peak count across histones")
```

```{r peak ct grouptime lq1e2b1e1}
all_peak_var %>% 
  ggplot(., aes (x=Timepoint, y = Count, fill = Histone_Mark))+
  geom_boxplot()+
   ylab("Count")+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle("Peak count across histones")
```

### Peak Calling for Q=0.01 and Broad=0.5 with No Lambda

#### Data Initialization
```{r peak data nlq1e2b5e}
peak_ct <- read_delim("data/peaks/peaks_cts_nlq1e2b5e1.txt", delim = "\t")
H3K27ac_peaks <- read_delim("data/peaks/H3K27ac_final_results.tsv",delim = "\t")
H3K27me3_peaks <- read_delim("data/peaks/H3K27me3_nlq1e2b5e1_results.tsv",delim = "\t")
H3K36me3_peaks <- read_delim("data/peaks/H3K36me3_nlq1e2b5e1_results.tsv",delim = "\t")
H3K9me3_peaks <- read_delim("data/peaks/H3K9me3_nlq1e2b5e1_results.tsv",delim = "\t")

all_peak_var <- rbind(H3K27ac_peaks, H3K27me3_peaks, H3K36me3_peaks, H3K9me3_peaks)

all_peak_var <- all_peak_var %>%
  dplyr::select(Sample, Total_Reads, Fragments, Reads_in_Peaks, FRiP) %>%
  left_join(.,sampleinfo, by=c("Sample"="Library ID")) %>%
  left_join(.,peak_ct, by=c("Sample"="Sample"))
all_peak_var <- all_peak[(!all_peak$Treatment %in% "5FU"),]
all_peak_var <- all_peak_var[(!all_peak_var$Histone_Mark %in% "H3K27ac"),]
```

#### Peak Visualization
```{r peak ct nlq1e2b5e, fig.width=14}
all_peak_var %>% 
   ggplot(.,aes(x=Sample, y=Count,fill=Histone_Mark))+
   geom_col()+
   ylab("Counts")+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle("Peak number for all samples")+ 
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  scale_y_continuous( expand = expansion(mult = c(0, .1)))
```

```{r peak ct box nlq1e2b5e}
all_peak_var %>% 
  ggplot(., aes (x=Histone_Mark, y = Count, fill = Histone_Mark))+
  geom_boxplot()+
   ylab("Count")+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle("Peak count across histones")
```

```{r peak ct grouptx nlq1e2b5e}
all_peak_var %>% 
  ggplot(., aes (x=Treatment, y = Count, fill = Histone_Mark))+
  geom_boxplot()+
   ylab("Count")+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle("Peak count across histones")
```

```{r peak ct grouptime nlq1e2b5e}
all_peak_var %>% 
  ggplot(., aes (x=Timepoint, y = Count, fill = Histone_Mark))+
  geom_boxplot()+
   ylab("Count")+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle("Peak count across histones")
```

### Peak Calling for Q=0.01 and Broad=0.5 with Lambda

#### Data Initialization
```{r peak data lq1e2b5e1}
peak_ct_var <- read_delim("data/peaks/peaks_cts_lq1e2b5e1.txt", delim = "\t")
H3K27ac_peaks <- read_delim("data/peaks/H3K27ac_final_results.tsv",delim = "\t")
H3K27me3_peaks <- read_delim("data/peaks/H3K27me3_lq1e2b5e1_results.tsv",delim = "\t")
H3K36me3_peaks <- read_delim("data/peaks/H3K36me3_lq1e2b5e1_results.tsv",delim = "\t")
H3K9me3_peaks <- read_delim("data/peaks/H3K9me3_lq1e2b5e1_results.tsv",delim = "\t")

all_peak_var <- rbind(H3K27ac_peaks, H3K27me3_peaks, H3K36me3_peaks, H3K9me3_peaks)

all_peak_var <- all_peak_var %>%
  dplyr::select(Sample, Total_Reads, Fragments, Reads_in_Peaks, FRiP) %>%
  left_join(.,sampleinfo, by=c("Sample"="Library ID")) %>%
  left_join(.,peak_ct, by=c("Sample"="Sample"))
all_peak_var <- all_peak_var[(!all_peak_var$Treatment %in% "5FU"),]
all_peak_var <- all_peak_var[(!all_peak_var$Histone_Mark %in% "H3K27ac"),]
```

#### Peak Visualization
```{r peak ct lq1e2b5e1, fig.width=14}
all_peak_var %>% 
   ggplot(.,aes(x=Sample, y=Count,fill=Histone_Mark))+
   geom_col()+
   ylab("Count")+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle("Peak number for all samples")+ 
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  scale_y_continuous( expand = expansion(mult = c(0, .1)))
```

```{r peak ct box lq1e2b5e1}
all_peak_var %>% 
  ggplot(., aes (x=Histone_Mark, y = Count, fill = Histone_Mark))+
  geom_boxplot()+
   ylab("Count")+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle("Peak count across histones")
```

```{r peak ct grouptx lq1e2b5e1}
all_peak_var %>% 
  ggplot(., aes (x=Treatment, y = Count, fill = Histone_Mark))+
  geom_boxplot()+
   ylab("Count")+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle("Peak count across histones")
```

```{r peak ct grouptime lq1e2b5e1}
all_peak_var %>% 
  ggplot(., aes (x=Timepoint, y = Count, fill = Histone_Mark))+
  geom_boxplot()+
   ylab("Count")+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle("Peak count across histones")
```

### Peak Calling for Q=0.005 and Broad=0.01 with No Lambda

#### Data Initialization
```{r peak data nlq5e3b1e2}
peak_ct <- read_delim("data/peaks/peaks_cts_nlq5e3b1e2.txt", delim = "\t")
H3K27ac_peaks <- read_delim("data/peaks/H3K27ac_final_results.tsv",delim = "\t")
H3K27me3_peaks <- read_delim("data/peaks/H3K27me3_nlq5e3b1e2_results.tsv",delim = "\t")
H3K36me3_peaks <- read_delim("data/peaks/H3K36me3_nlq5e3b1e2_results.tsv",delim = "\t")
H3K9me3_peaks <- read_delim("data/peaks/H3K9me3_nlq5e3b1e2_results.tsv",delim = "\t")

all_peak_var <- rbind(H3K27ac_peaks, H3K27me3_peaks, H3K36me3_peaks, H3K9me3_peaks)

all_peak_var <- all_peak_var %>%
  dplyr::select(Sample, Total_Reads, Fragments, Reads_in_Peaks, FRiP) %>%
  left_join(.,sampleinfo, by=c("Sample"="Library ID")) %>%
  left_join(.,peak_ct, by=c("Sample"="Sample"))
all_peak_var <- all_peak_var[(!all_peak_var$Treatment %in% "5FU"),]
all_peak_var <- all_peak_var[(!all_peak_var$Histone_Mark %in% "H3K27ac"),]
```

#### Peak Visualization
```{r peak ct nlq5e3b1e2, fig.width=14}
all_peak_var %>% 
   ggplot(.,aes(x=Sample, y=Count,fill=Histone_Mark))+
   geom_col()+
   ylab("Count")+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle("Peak number for all samples")+ 
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  scale_y_continuous( expand = expansion(mult = c(0, .1)))
```

```{r peak ct box nlq5e3b1e2}
all_peak_var %>% 
  ggplot(., aes (x=Histone_Mark, y = Count, fill = Histone_Mark))+
  geom_boxplot()+
   ylab("Count")+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle("Peak count across histones")
```

```{r peak ct grouptx nlq5e3b1e2}
all_peak_var %>% 
  ggplot(., aes (x=Treatment, y = Count, fill = Histone_Mark))+
  geom_boxplot()+
   ylab("Count")+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle("Peak count across histones")
```

```{r peak ct grouptime nlq5e3b1e2}
all_peak_var %>% 
  ggplot(., aes (x=Timepoint, y = Count, fill = Histone_Mark))+
  geom_boxplot()+
   ylab("Count")+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle("Peak count across histones")
```

### Peak Calling for Q=0.005 and Broad=0.01 with Lambda

#### Data Initializationx
```{r peak data lq5e3b1e2}
peak_ct <- read_delim("data/peaks/peaks_cts_lq5e3b1e2.txt", delim = "\t")
H3K27ac_peaks <- read_delim("data/peaks/H3K27ac_final_results.tsv",delim = "\t")
H3K27me3_peaks <- read_delim("data/peaks/H3K27me3_lq5e3b1e2_results.tsv",delim = "\t")
H3K36me3_peaks <- read_delim("data/peaks/H3K36me3_lq5e3b1e2_results.tsv",delim = "\t")
H3K9me3_peaks <- read_delim("data/peaks/H3K9me3_lq5e3b1e2_results.tsv",delim = "\t")

all_peak_var <- rbind(H3K27ac_peaks, H3K27me3_peaks, H3K36me3_peaks, H3K9me3_peaks)

all_peak_var <- all_peak_var %>%
  dplyr::select(Sample, Total_Reads, Fragments, Reads_in_Peaks, FRiP) %>%
  left_join(.,sampleinfo, by=c("Sample"="Library ID")) %>%
  left_join(.,peak_ct, by=c("Sample"="Sample"))
all_peak_var <- all_peak_var[(!all_peak_var$Treatment %in% "5FU"),]
all_peak_var <- all_peak_var[(!all_peak_var$Histone_Mark %in% "H3K27ac"),]
```

#### Peak Visualization
```{r peak ct lq5e3b1e2, fig.width=14}
all_peak_var %>% 
   ggplot(.,aes(x=Sample, y=Count,fill=Histone_Mark))+
   geom_col()+
   ylab("Count")+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle("Peak number for all samples")+ 
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  scale_y_continuous( expand = expansion(mult = c(0, .1)))
```

```{r peak ct box lq5e3b1e2}
all_peak_var %>% 
  ggplot(., aes (x=Histone_Mark, y = Count, fill = Histone_Mark))+
  geom_boxplot()+
   ylab("Count")+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle("Peak count across histones")
```

```{r peak ct grouptx lq5e3b1e2}
all_peak_var %>% 
  ggplot(., aes (x=Treatment, y = Count, fill = Histone_Mark))+
  geom_boxplot()+
   ylab("Count")+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle("Peak count across histones")
```

```{r peak ct grouptime lq5e3b1e2}
all_peak_var %>% 
  ggplot(., aes (x=Timepoint, y = Count, fill = Histone_Mark))+
  geom_boxplot()+
   ylab("Count")+
   theme_classic()+
  # facet_wrap(~histone)+
  ggtitle("Peak count across histones")
```

#### Tagging Questionable Libraries by FRiP
```{r frip tag}
questionable_frip = all_peak[(all_peak$FRiP < 0.02),]
questionable_frip
```

```{r rename}
all_peak <- all_peak %>%
  mutate(Sample=paste0(Individual,"_",Treatment,"_",Timepoint,"_",Histone_Mark))
```

### Feature Counts
```{r feature_cts}
featurects_merged <- read_delim("data/peaks/H3K27ac_merged_counts.txt", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE, skip = 1)
featurects_cluster <- read_delim("data/peaks/H3K27ac_cluster_counts.txt", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE, skip = 1)
featurects_iter <- read_delim("data/peaks/H3K27ac_iter_counts.txt", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE, skip = 1)
rename_list <- sampleinfo %>% 
  mutate(stem= "_nobl.bam") %>% 
  mutate(prefix=paste0("/scratch/10819/styu/MW_multiQC/peaks/",Histone_Mark,"/",Treatment,"/",Timepoint,"/")) %>%
  mutate(oldname=paste0(prefix,`Library ID`,"/",`Library ID`,stem)) %>% 
  mutate(newname=paste0(Individual,"_",Treatment,"_",Timepoint,"_",Histone_Mark)) %>% 
  dplyr::select(oldname,newname)
rename_vec <- setNames(rename_list$newname, rename_list$oldname)
names(featurects_merged)[names(featurects_merged) %in% names(rename_vec)] <- rename_vec[names(featurects_merged)[names(featurects_merged) %in% names(rename_vec)]]
names(featurects_cluster)[names(featurects_cluster) %in% names(rename_vec)] <- rename_vec[names(featurects_cluster)[names(featurects_cluster) %in% names(rename_vec)]]
names(featurects_iter)[names(featurects_iter) %in% names(rename_vec)] <- rename_vec[names(featurects_iter)[names(featurects_iter) %in% names(rename_vec)]]
```

### H3K27ac Count Analysis
```{r H3K27ac HeatMap 1, fig.width=14,fig.height=14}
H3K27ac_merged_raw <- featurects_merged %>% 
  dplyr::select(Geneid,contains("Ind")) %>% 
  column_to_rownames("Geneid") %>% 
  as.matrix()

H3K27ac_cluster_raw <- featurects_cluster %>% 
  dplyr::select(Geneid,contains("Ind")) %>% 
  column_to_rownames("Geneid") %>% 
  as.matrix()

H3K27ac_iter_raw <- featurects_iter %>% 
  dplyr::select(Geneid,contains("Ind")) %>% 
  column_to_rownames("Geneid") %>% 
  as.matrix()

H3K27ac_merged_cor <- featurects_merged %>% 
  dplyr::select(Geneid,contains("Ind")) %>% 
  column_to_rownames("Geneid") %>% 
  cpm(., log = TRUE) %>% 
  cor()

H3K27ac_cluster_cor <- featurects_cluster %>% 
  dplyr::select(Geneid,contains("Ind")) %>% 
  column_to_rownames("Geneid") %>% 
  cpm(., log = TRUE) %>% 
  cor()

H3K27ac_iter_cor <- featurects_iter %>% 
  dplyr::select(Geneid,contains("Ind")) %>% 
  column_to_rownames("Geneid") %>% 
  cpm(., log = TRUE) %>% 
  cor()

annomat <- data.frame(sample=colnames(H3K27ac_merged_cor)) %>% 
  separate_wider_delim(sample,delim="_",names=c("Ind","Treatment","Timepoint",NA),cols_remove = FALSE) %>% 
  mutate(Treatment=factor(Treatment, levels = c("VEH","5FU","DOX")),
         Timepoint=factor(Timepoint, levels =c("24T","24R","144R"))) %>% 
  column_to_rownames("sample")
heatmap_first <- ComplexHeatmap::HeatmapAnnotation(df = annomat)

Heatmap(H3K27ac_merged_cor, 
        top_annotation = heatmap_first,
        column_title="Unfiltered log2cpm H3K27ac")
```

```{r H3K27ac HeatMap 2, fig.width=14,fig.height=14}
annomat <- data.frame(sample=colnames(H3K27ac_cluster_cor)) %>% 
  separate_wider_delim(sample,delim="_",names=c("Ind","Treatment","Timepoint",NA),cols_remove = FALSE) %>% 
  mutate(Treatment=factor(Treatment, levels = c("VEH","5FU","DOX")),
         Timepoint=factor(Timepoint, levels =c("24T","24R","144R"))) %>% 
  column_to_rownames("sample")
heatmap_first <- ComplexHeatmap::HeatmapAnnotation(df = annomat)

Heatmap(H3K27ac_cluster_cor, 
        top_annotation = heatmap_first,
        column_title="Unfiltered log2cpm H3K27ac")
```

```{r H3K27ac HeatMap 3, fig.width=14,fig.height=14}
annomat <- data.frame(sample=colnames(H3K27ac_iter_cor)) %>% 
  separate_wider_delim(sample,delim="_",names=c("Ind","Treatment","Timepoint",NA),cols_remove = FALSE) %>% 
  mutate(Treatment=factor(Treatment, levels = c("VEH","5FU","DOX")),
         Timepoint=factor(Timepoint, levels =c("24T","24R","144R"))) %>% 
  column_to_rownames("sample")
heatmap_first <- ComplexHeatmap::HeatmapAnnotation(df = annomat)

Heatmap(H3K27ac_iter_cor, 
        top_annotation = heatmap_first,
        column_title="Unfiltered log2cpm H3K27ac")
```

#### H3K27ac Fragments
```{r frag ct, fig.width=14}
all_peak %>%
  mutate(Fragments=Fragments/1000000) %>% 
  ggplot(., aes(x=interaction(Individual,Treatment,Timepoint), y=Fragments, fill=Treatment, group = Treatment))+
  geom_col()+
  geom_hline(yintercept =5)+
  geom_text(aes(y = 0,label = Sample), vjust = 0.2, size = 3, angle = 90)+
  theme_classic()+
  facet_wrap(~Histone_Mark)+
  ggtitle("Fragment count by histone and sample")+
  ylab("Count of Fragments * 10^6")+
  xlab("Samples")+
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  scale_y_continuous( expand = expansion(mult = c(0, .1)))
```

```{r frag frip, fig.width=14}
all_peak %>%
  mutate(FRiP=FRiP * 100) %>% 
  ggplot(., aes(x=interaction(Individual,Treatment,Timepoint), y=FRiP, fill=Treatment, group = Treatment))+
  geom_col()+
  geom_hline(yintercept =5)+
  geom_text(aes(y = 0,label = Sample), vjust = 0.2, size = 3, angle = 90)+
  theme_classic()+
  facet_wrap(~Histone_Mark)+
  ggtitle("Fragment count by histone and sample")+
  ylab("Count of Fragments * 10^6")+
  xlab("Samples")+
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  scale_y_continuous( expand = expansion(mult = c(0, .1)))
```

### H3K27me3 Count Analysis

### H3K36me3 Count Analysis

### H3K9me3 Count Analysis


